backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java

```
@@ -1497,6 +1506,265 @@ public Bundle createFhirBundle(CreateFhirBundleProps props) {
+                        patientInput.getCity(),
+                        patientInput.getState(),
+                        patientInput.getZipCode(),
+                        patientInput.getCounty()))
+                .country(patientInput.getCountry())
+                .race(patientInput.getRace())
+                .ethnicity(patientInput.getEthnicity())
+                .tribalAffiliations(Collections.singletonList(patientInput.getTribalAffiliation()))
+                .build());
+
+    StreetAddress facilityAddress =
+        new StreetAddress(
+            facility.getStreet(),
+            facility.getStreetTwo(),
+            facility.getCity(),
+            facility.getState(),
+            facility.getZipCode(),
+            facility.getCounty());
+
+    var testingLab =
+        convertToOrganization(
+            facility.getInternalId().toString(),
+            facility.getName(),
+            facility.getClia(),
+            facility.getPhone(),
+            facility.getEmail(),
+            facilityAddress,
+            facility.getCountry());
+
+    StreetAddress providerAddress =
+        new StreetAddress(
+            providerInput.getStreet(),
+            providerInput.getStreetTwo(),
+            providerInput.getCity(),
+            providerInput.getState(),
+            providerInput.getZipCode(),
+            providerInput.getCounty());
+
+    var practitioner =
+        convertToPractitioner(
+            providerInput.getInternalId().toString(),
+            new PersonName(
+                providerInput.getFirstName(),
+                providerInput.getMiddleName(),
+                providerInput.getFirstName(),
+                providerInput.getSuffix()),
+            providerInput.getPhone(),
+            providerAddress,
+            providerInput.getCountry(),
+            providerInput.getNpi());
+
+    ZonedDateTime specimenCollectionDate =
```

Are specimen dates a required field? If so, should we throw an error here if they do not exist?

---

We'll want to confirm with product, but based on the current bulk upload guide, specimen collection date and specimen received date are optional fields.